app/client/cli/servicer.go

```diff
@@ -0,0 +1,243 @@
+
+				return nil
+			},
+		},
+	}
+
+	return cmds
+}
+
+// TODO: add a cli command for fetching sessions
+// validateServicer returns the servicer specified by the <servicer> argument.
+// It validates that the <servicer> is the address of a servicer that is active in the current session.
+func validateServicer(ctx context.Context, session *rpc.Session, servicerAddress string) (*rpc.ProtocolActor, error) {
+	for _, s := range session.Servicers {
+		if s.Address == servicerAddress {
+			return &s, nil
+		}
+	}
+
+	// ADDTEST: cover with gherkin tests
+	return nil, fmt.Errorf("Error getting servicer: address %s does not match any servicers in the session %d", servicerAddress, session.SessionNumber)
+}
+
+func getCurrentSession(ctx context.Context, appAddress, chain string) (*rpc.Session, error) {
+	// CONSIDERATION: passing 0 as the height value to get the current session seems more optimal than this.
+	currentHeight, err := getCurrentHeight(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session: %w", err)
+	}
+
+	req := rpc.SessionRequest{
+		AppAddress: appAddress,
+		Chain:      chain,
+		// TODO(#697): Geozone
+		SessionHeight: currentHeight,
+	}
+
+	client, err := rpc.NewClientWithResponses(remoteCLIURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session for app/chain/height: %s/%s/%d: %w", appAddress, chain, currentHeight, err)
+	}
+
+	resp, err := client.PostV1ClientGetSessionWithResponse(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session with request %v: %w", req, err)
+	}
+
+	// IMPROVE: move the HTTP response processing code to a separate function to enable reuse.
+	if resp.HTTPResponse.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Error getting current session: Unexpected status code %d for request %v", resp.HTTPResponse.StatusCode, req)
+	}
+
+	if resp.JSON200 == nil {
+		return nil, fmt.Errorf("Error getting current session: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200, nil
+}
+
+// REFACTOR: reuse this function in all the query commands
+func getCurrentHeight(ctx context.Context) (int64, error) {
+	client, err := rpc.NewClientWithResponses(remoteCLIURL)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	resp, err := client.GetV1QueryHeightWithResponse(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	if resp.HTTPResponse.StatusCode != http.StatusOK {
+		return 0, fmt.Errorf("Error getting current height: Unexpected status code %d", resp.HTTPResponse.StatusCode)
+	}
+
+	if resp.JSON200 == nil {
+		return 0, fmt.Errorf("Error getting current height: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200.Height, nil
+}
+
+// IMPROVE: [K8s][LocalNet] Publish Servicer(s) Host and Port as env. vars in K8s: similar to Validators: #823
+// REFACTOR: move package-level variables (e.g. remoteCLIURL) to a cli object and consider storing it in the context
```

Comment:
```suggestion
// CONSIDERATION: move package-level variables (e.g. remoteCLIURL) to a cli object and consider storing it in the context
```

@bryanchriswhite recently refactored it to consolidate and export them from on eplace

Reply:
Thanks, fixed.